### PR TITLE
Remove mapToWorld() in navfn_planner.cpp

### PR DESCRIPTION
**Pull request by joeldushouyu**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | (https://github.com/ros-planning/navigation2/issues/3669) |
| Primary OS tested on | Ubuntu|
| Robotic platform tested on | gazebo simulation |

---

## Description of contribution in a few bullet points
* replace mapToWorld() by using the mapToWorld() in costmap_2d
* replace worldToMap() by using the worldToMap() in costmap_2d

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_navfn_planner/include/nav2_navfn_planner/navfn_planner.hpp [hunk 1]
```diff
@@ -166,15 +166,6 @@ class NavfnPlanner : public nav2_core::GlobalPlanner
    */
   bool worldToMap(double wx, double wy, unsigned int & mx, unsigned int & my);
```
Comment (SteveMacenski) [inline nav2_navfn_planner/include/nav2_navfn_planner/navfn_planner.hpp:167]: We may as well replace this function while we're at it!

File: nav2_navfn_planner/src/navfn_planner.cpp [hunk 2]
```diff
@@ -417,7 +417,7 @@ NavfnPlanner::getPlanFromPotential(
   for (int i = len - 1; i >= 0; --i) {
     // convert the plan to world coordinates
     double world_x, world_y;
-    mapToWorld(x[i], y[i], world_x, world_y);
+    this->costmap_->mapToWorld(x[i], y[i], world_x, world_y);
```
Comment (SteveMacenski) [inline nav2_navfn_planner/src/navfn_planner.cpp:420]: No need for `this->`

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: nav2_navfn_planner/include/nav2_navfn_planner/navfn_planner.hpp (hunk 1 above)
Comment (joeldushouyu) [reply nav2_navfn_planner/include/nav2_navfn_planner/navfn_planner.hpp:167]: Should I keep the RCLCPP_ERROR log? 
Or does the implementation always assume successful conversion from world coordinates to map?

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
@joeldushouyu sorry for the delay

```

-  if(!costmap_->worldToMap(wx, wy, mx, my)){
+  if (!costmap_->worldToMap(wx, wy, mx, my)) {
@@ -270 +270 @@
-  if(!costmap_->worldToMap(wx, wy, mx, my)){
+  if (!costmap_->worldToMap(wx, wy, mx, my)) {
@@ -402 +402 @@
-  if(!costmap_->worldToMap(wx, wy, mx, my)){
+  if (!costmap_->worldToMap(wx, wy, mx, my)) {
```

You have some linting problems. Also, please pull in `main`, there are a **large** number of tests failing which I think should be fine if you do so. After that, any tests failing may be legit things to look into for this change. There are a few new failures to me in the planning system tests which makes me think perhaps there is a legit problem here (or at  the very least, update to tests needed)

**Comment by joeldushouyu** [comment]
> @joeldushouyu sorry for the delay
> 
> ```
> 
> -  if(!costmap_->worldToMap(wx, wy, mx, my)){
> +  if (!costmap_->worldToMap(wx, wy, mx, my)) {
> @@ -270 +270 @@
> -  if(!costmap_->worldToMap(wx, wy, mx, my)){
> +  if (!costmap_->worldToMap(wx, wy, mx, my)) {
> @@ -402 +402 @@
> -  if(!costmap_->worldToMap(wx, wy, mx, my)){
> +  if (!costmap_->worldToMap(wx, wy, mx, my)) {
> ```
> 
> You have some linting problems. Also, please pull in `main`, there are a **large** number of tests failing which I think should be fine if you do so. After that, any tests failing may be legit things to look into for this change. There are a few new failures to me in the planning system tests which makes me think perhaps there is a legit problem here (or at the very least, update to tests needed)

I have made changes to fix the lining issue and pull on the main branch.
It did seem that the change caused a few errors.  

Should I try to dig into the error? Or will it be best to leave them to you to take a look at?  @SteveMacenski

**Comment by SteveMacenski** [comment]
I’d look into them. Those are new failures so definitely as a result of changes

**Comment by joeldushouyu** [comment]
I would be glad to take a look into it. 

I wonder if there is a way that I can trigger those tests locally on my laptop? By the way, I am on ubuntu 20.04 and had galactic version of ros2 installed; not sure if that will matter.

Thanks

**Comment by SteveMacenski** [comment]
Colcon test will execute the tests locally

**Comment by SteveMacenski** [comment]
@joeldushouyu any updates :-)

**Comment by joeldushouyu** [comment]
Oops. I am sorry! I totally forgot about it. 
I was busy preparing for my GRE exam that is coming next week.
I will be able to work on it next week after my exam.  Sorry about that!

However, if you feel that this is taking too long. I am welcome to have others take a look at those errors.

**Comment by SteveMacenski** [comment]
See follow up in https://github.com/ros-planning/navigation2/issues/3669

**Pull request closed without merging**